tests/virt_autotest/virt_utils.pm

```diff
@@ -691,11 +691,14 @@ sub perform_guest_restart {
 #Please refer to virt_logs_collector.sh and fetch_logs_from_guest.sh in data/virt_autotest for their detailed functionality, implementation and usage
 sub collect_host_and_guest_logs {
     my ($guest_wanted, $host_extra_logs, $guest_extra_logs) = @_;
+    $guest_wanted     //= '';
+    $host_extra_logs  //= '';
+    $guest_extra_logs //= '';
 
     my $logs_collector_script_url = data_url("virt_autotest/virt_logs_collector.sh");
     script_output("curl -s -o ~/virt_logs_collector.sh $logs_collector_script_url", 180, type_command => 0, proceed_on_failure => 0);
     save_screenshot;
-    script_output("chmod +x ~/virt_logs_collector.sh && ~/virt_logs_collector.sh -l \"$host_extra_logs\" -g \"$guest_wanted\" -e \"$guest_extra_logs\"", 1800, type_command => 1, proceed_on_failure => 1);
+    script_output("chmod +x ~/virt_logs_collector.sh && ~/virt_logs_collector.sh -l \"$host_extra_logs\" -g \"$guest_wanted\" -e \"$guest_extra_logs\"", 3600, type_command => 1, proceed_on_failure => 1);
```

Comment:
On osd, the SCALE is 3x, which means 1 hour timeout will be actually 3 hours. Is this really needed to be that long? In extreme situations where this script_output hang, this 3h will be likely to be wasted most. 

And do not forget openqa parameter,  max_job_time or max_test_time,  which will terminate the job if overall job time exceeds that. So the parameter is likely to need enlarging too if needed(we have been complained a lot by tools team that our job timeout is too long than other groups(~6 hours+)), otherwise in extreme situations, the latter post_fail_hook scripts may not have time to be executed before job terminated.

Reply:
timeout for script_output is just timeout for script_output which has nothing to do with scale. And there is no need to extend job timeout. Additionally, generally speaking, job timeout does not needs to be the sum of all timeouts in job, because not all extreme conditions happen all together.

Reply:
@okurz I guess you have more professional input for this timeout thing?

Reply:
It is correct that the specified timeout value will be scaled by the parameter "TIMEOUT_SCALE", see https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L916 . And I agree with @alice-suse that it is a good approach to select a value for MAX_JOB_TIME which is bigger than the sum of all internal timeouts within one execution path. Of course it is unlikely that multiple bad condititions come together. This can be considered. The main example where this was important in the path was a complete system installation which was taking long due to limited network connection which can lead to increased time in multiple steps, e.g. initial repository refreshing, SCC registration, downloading packages. This led to incomplete jobs which prevents a useful label carry over and without helpful logs the investigation for test reviewers was more difficult. MAX_JOB_TIME should be seen as last-resort to catch ill-designed test code, e.g. having infinite loops. In your case I think bumping MAX_JOB_TIME accordingly to cover the increased timeout is a good and safe approach as @alice-suse suggested. About the remark that "job timeout is too long than other groups": Of course it is good if in case of any failure jobs terminate as soon as possible to pin-point to the actual error source and save time.

Reply:
It seems that I referred to some mysterious documentation :)   Thanks for @okurz  your detailed explanation and useful direction https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L916. Really impressed.

Reply:
Additionally this function is a standalone. So keep 3600 is also still correct from my perspective. I do not expect everyone to use scale always. But if you @alice-suse  insist to change it, please give your value, then I will change it (I can not figure one out by myself). But you need to keep in mind that 3660 is ideal final value.

Reply:
@waynechen55 Regarding timeout value, my suggestion is to use it as large as needed, but not wasted, so you decide it according to this test step's real need. 
If 3600 is final value(after scale), then I suggest to  change code to be timeout value as 3600/get_var('TIMEOUT_SCALE', 1). For OSD, it will be 1200, which is even smaller than your previous value 1800, then you do not need to enlarge MAX_JOB_TIME(or can even be a little bit smaller, up to you). For most people's local openqa env, likely no scale(or 1x), then 3600 is enough too.

Reply:
Done